lib/TH/THFilePrivate.h

```diff
@@ -32,6 +40,7 @@ struct THFileVTable
     size_t (*writeLong)(THFile *self, long *data, size_t n);
     size_t (*writeFloat)(THFile *self, float *data, size_t n);
     size_t (*writeDouble)(THFile *self, double *data, size_t n);
+    size_t (*writeHalf)(THFile *self, half *data, size_t n);
```

Comment:
@gchanan: the intent was to not define any functions except copy/init and file I/O. So that one can just write a pipeline using CUDNN with half and avoid all conversions. I suggest we revisit that if/when I manage to get NATIVE_HALF working ?